### ISSUE TITLE
deprecation warning in use of x25519

### ISSUE THREAD

**Issue by warner** [plizonczyk/noiseprotocol#32]
I noticed the following deprecation warning when running a magic-wormhole test:

```
.../.tox/py35/lib/python3.5/site-packages/cryptography/hazmat/backends/openssl/x25519.py:35: CryptographyDeprecationWarning: public_bytes now requires encoding and format arguments. Support for calling without arguments will be removed in cryptography 2.7
```

I think this is coming from `noise/backends/default/diffie_hellmans.py`, where it does `return KeyPair25519(private_key, public_key, public_key.public_bytes())`. I'm guessing that `cryptography` now wants additional arguments for the `public_bytes()` call to specify the encoding.

The `x25519` module is also imported by `noise/backends/default/keypairs.py`, so maybe something will need to be patched in there too.

Cc'ing @reaperhulk to get a sense for how urgent this might or might not be.

**Comment by jcoetzee** [comment]
Any idea when this will be merged? This breaks running the example code (and assuming basicaly everything else?) when using Cryptography 2.7 (which pip installs by default when running `pip install noiseprotocol`).

**Comment by plizonczyk** [comment]
Fixed with version 0.3.1.

Thank you for contribution @warner !!! :)
And sorry for keeping you all waiting.
